### DISCUSSION TITLE
Leveling Timer

### DISCUSSION THREAD

**Discussion by PergonGER** [Lailloken/Exile-UI#124] [Ideas & Feedback]
Hey,
i think a leveling timer would be great addition for the leveling part of this tool.

- "Track time while window is focussed" option
- a pause/continue timer button (for trades for example)
- track timer for each zone and act
- add a possibility to compare it to old runs (for league start comparison ie)

**Comment by Lailloken** [comment]
Sounds reasonable. I could enable acts in the mapping tracker since it does pretty much everything you're asking for: doesn't track alt-tabbed time, pauses in the hideout/towns, creates logs that can be viewed and exported. I'm not sure I would go as far as tracking the campaign zone by zone, but rather act by act, because many areas have backtracking and putting that into a visually pleasant log sounds like a nightmare. But that's just my spontaneous reaction, I'll put more thought into it later.

Unfortunately, I don't think I'll be able to get it ready for Friday because I'm working on something that I think is pretty big for league-starts. I'm expanding the item-info tooltip so that it automatically compares the item you found with the item you have equipped at the moment, like +X life and -Y resist, etc. It's too early right now to say when that's done, and then I'll also have to implement the new item and skill tree data, as well as the shuffled atlas.

Thanks for your suggestion, I'll try my best to get it ready for Friday.

**Reply by PergonGER** [reply]
Wow, awesome. More than i expected. :) Tracking acts is more than enough.

The item comparison sounds great and you should definetly focus on that one. I really dig your tool. Best one around (beside trade addon, which is mandatory), adds so much QoL to the game.

**Reply by bk-sin** [reply]
Tracking acts or the total time of the run it would be great ❤

**Comment by Lailloken** [comment]
Took some time to find a solution I'm happy with (integrating it into the mapping tracker ended up being a worse solution than I had initially thought). Sneak peek:
![image](https://user-images.githubusercontent.com/61888437/229816618-d865845f-680b-4153-8093-64f8b14c8496.png)

It will have basic logging to CSV-files and maybe some UI to compare the current run/acts to averages.
